tauri: migrating to V2

xTVaser opened the pull request:
Fixes #529
Fixes #581
Fixes #224

~~May potentially solve #581 as it uses a newer version of webkit2gtk -- im currently convinced the UI rendering bug is the fault of webkitgtk2, and i tested last night that on newer versions (i used GNOME Web) it doesn't manifest.~~
- Confirmed to solve the issue, though it causes others 😄 https://bugs.launchpad.net/ubuntu/+source/webkit2gtk/+bug/2081166

Before merging:
- [x] make sure the release process works (test on a fork)
  - [x] don't forget to do #224
- [x] test on all operating systems
  - [x] windows
  - [x] macos
  - [x] linux
    - tested here https://github.com/open-goal/launcher/issues/701#issuecomment-2764665002

This is blocked for the following reasons:
- Linux
  - https://github.com/tauri-apps/tauri/issues/11171 (our app is multi-window and it doesn't seem to work properly, i had to temporarily make it a single window app to get it to even launch)
  - https://github.com/tauri-apps/tauri/issues/10626 (bad webkit version out there? upstream ubuntu bug report https://bugs.launchpad.net/ubuntu/+source/webkit2gtk/+bug/2081166)

xTVaser committed:
tauri: run `tauri migrate`

xTVaser committed:
tauri: add all plugins

xTVaser committed:
tauri: mostly finished migration, app builds and runs again

xTVaser committed:
lint: formatting

trippjoe commented:
Both linux related tauri bugs have been closed.

Milkfat32 commented:
> https://github.com/tauri-apps/tauri/issues/10626 (bad webkit version out there? upstream ubuntu bug report https://bugs.launchpad.net/ubuntu/+source/webkit2gtk/+bug/2081166)

If the Linux version is to be released only as a Flatpak, this is not an issue as the webkit version is controlled in the Flatpak sandbox. This is done through the runtime sdk selection and not the system libraries.

xTVaser commented:
It was an issue because you couldn't develop or debug the app on ubuntu linux.

Milkfat32 committed:
Linux fixes for tauri-v2 (#660)

Updated tauri packages
Added tauri.linux.conf to disable tauri updater

Also went with default flatpak install directory data/ folder for
installation. Button on splash is replaced by Continue after locale
select.
Removed the Installation Directory setting for Linux build. I didn't
think it should be able to be changed? If someone changed it to some
directory in /home/ without giving access through flatseal/terminal it
would throw errors so I assumed this was the best solution

Added icons/desktop file since that should be upstream according to
flathub maintainers. Once v2 is released I will update the flathub repo,
since flathub bot builds from the git releases on this repo. The appdata
file does not work as intended yet in the ci, have only gotten that to
update properly on the flathub repo

Ping me on Discord and I will create a request to add Vaser/tripp as
maintainer, since I don't own the repo on flathub.

---------

Co-authored-by: Milkfat32 <Milkfat32@users.noreply.github.com>

xTVaser committed:
Merge branch 'main' into v/tauri-v2

xTVaser committed:
v2: a bunch of merge conflict resolution changes

xTVaser committed:
v2/ci: dont require private key in non-release builds

xTVaser committed:
lint: formatting

xTVaser commented on src-tauri/tauri.linux.conf.json:
```diff
@@ -0,0 +1,7 @@
+            "active": false
+        }
+    }
+}
```
why we we disabling the updater in linux?

Milkfat32 replied:
Updates to flatpak is handled by the flatpak package manager. For diy distros like gentoo/arch this is done with `flatpak update`, for others this is usually in some software manager.

xTVaser commented on src-tauri/tauri.conf.json:
```diff
@@ -1,82 +1,75 @@
+      "msi",
+      "app",
+      "dmg"
+    ],
```
has no flatpak configuration, looks like we need to build the `deb`.  CI will have to similarly be updated to accommodate the new release asset.

Milkfat32 replied:
That should not be necessary. Flatpaks in flathub are built in the flathub repository, so the launcher has its own [repository](https://github.com/flathub/dev.opengoal.OpenGOAL) in flathub. I have a workflow that creates the offline mirrors for the rust and npm dependencies and I build from source. I believe building a flatpak file outside the repository simply creates a .flatpakref file that users have to manually install. In the flathub repository it shows up in software managers.

xTVaser replied:
I don't really like the idea of us 100% depending on flathub / external repos to handle our linux distribution.  From working on / observing [PCSX2](https://github.com/PCSX2/pcsx2/blob/master/.github/workflows/linux_build_flatpak.yml) they follow a model I much prefer, where they build the flatpak and publish it to flathub, that means the user can either download the flatpak directly from the pcsx2 website, or via flathub.

Milkfat32 replied:
Understandable. I did not know it was possible to build and publish outside the Flathub repositories. It does seem like a lot of extra work though, and I am not entirely sure how publishing to Flathub works. It seems like you need some kind of access token, which I don't know how to get other than perhaps asking for it? The documentation on this is very sparse.

I don't think it is that big a benefit to download the flatpak from the website, as users are generally conditioned to only seek software from repositories or through software managers

In my opinion this should be considered for the future, like it also seems the PCSX2 devs did as they moved from the flathub github repositories to self publishing, as to not stall the v2 branch for too long as I would have to do a lot more digging and asking around

trippjoe committed:
fixes for tauri v2 (#666)

misc changes i find while testing.

tauri changed the platform() function in v2, it's no longer async so we
dont need to await. this lets us remove it from the onMount function and
use it in the top level script. the platform() function also returns
"windows", "macos", or "linux" now in lieu of "win32" and "darwin". this
allowed me to simplify the logic in mods.ts.

added /gen folder to the gitignore

did a full pass through with cargo clippy to adhere to rust best practices

trippjoe commented on src-tauri/Cargo.toml:
bumping the MSRV "minimum supported rust version" was required to resolve a clippy warning

xTVaser committed:
Merge remote-tracking branch 'origin/main' into v/tauri-v2

xTVaser committed:
tauri: disable updater unless it's being released

xTVaser committed:
ci: update linux dependencies

xTVaser committed:
v2: various linux fixes

xTVaser committed:
tauri: don't build updater artifacts outside of releases

xTVaser committed:
update deps and lint

xTVaser committed:
tests: fix tests, disable component ones as testing-library doesnt seem to support svelte5

xTVaser committed:
tailwind: remove `require` causing the app to not launch

xTVaser committed:
ci: check and simplify release workflows a bit

xTVaser committed:
tauri: remove flatpak specific linux code from the frontend, breaks appimages, we aren't ready for flatpaks

xTVaser committed:
lint: formatting
